geode-core/src/main/java/org/apache/geode/management/internal/configuration/mutators/MemberConfigManager.java

```diff
@@ -59,39 +60,45 @@ public void delete(MemberConfig config, CacheConfig existing) {
 
-    Set<DistributedMember> members = cache.getDistributionManager().getDistributionManagerIds()
-        .stream().filter(m -> (filter.getId() == null || filter.getId().equals(m.getName())))
-        .map(DistributedMember.class::cast).collect(Collectors.toSet());
+    Set<DistributedMember> distributedMembers =
+        cache.getDistributionManager().getDistributionManagerIds()
+            .stream().filter(internalDistributedMember -> (filter.getId() == null
+                || filter.getId().equals(internalDistributedMember.getName())))
+            .map(DistributedMember.class::cast).collect(Collectors.toSet());
 
-    if (members.size() == 0) {
-      return results;
+    if (distributedMembers.size() == 0) {
+      return new ArrayList<>();
     }
 
-    for (DistributedMember member : members) {
-      if (member == getCoordinator()) {
-        coordinatorId = member.getId();
-      }
-    }
+    ArrayList<MemberInformation> memberInformation = getMemberInformation(distributedMembers);
+
+    return generateMemberConfigs(memberInformation);
+  }
 
+  private ArrayList<MemberInformation> getMemberInformation(Set<DistributedMember> members) {
     Execution execution = FunctionService.onMembers(members);
-    ResultCollector<?, ?> rc = execution.execute(new GetMemberInformationFunction());
+    ResultCollector<?, ?> resultCollector = execution.execute(new GetMemberInformationFunction());
+    return (ArrayList<MemberInformation>) resultCollector.getResult();
+  }
 
-    ArrayList<MemberInformation> output = (ArrayList<MemberInformation>) rc.getResult();
+  private List<MemberConfig> generateMemberConfigs(ArrayList<MemberInformation> memberInformation) {
 
-    for (MemberInformation mInfo : output) {
+    final String coordinatorId = getCoordinatorId();
+    List<MemberConfig> memberConfigs = new ArrayList<>();
+    for (MemberInformation information : memberInformation) {
       MemberConfig member = new MemberConfig();
-      member.setId(mInfo.getName());
-      member.setHost(mInfo.getHost());
-      member.setPid(mInfo.getProcessId());
-      member.setStatus(mInfo.getStatus());
-      member.setInitialHeap(mInfo.getInitHeapSize());
-      member.setMaxHeap(mInfo.getMaxHeapSize());
-
-      if (mInfo.isServer() && mInfo.getCacheServeInfo() != null) {
-        for (CacheServerInfo info : mInfo.getCacheServeInfo()) {
+      member.setId(information.getName());
+      member.setHost(information.getHost());
+      member.setPid(information.getProcessId());
+      member.setStatus(information.getStatus());
+      member.setInitialHeap(information.getInitHeapSize());
+      member.setMaxHeap(information.getMaxHeapSize());
+      member.setGroups(Arrays.asList(information.getGroups().split(",")));
+      member.setCoordinator(information.getId().equals(coordinatorId));
+
+      if (information.isServer() && information.getCacheServeInfo() != null) {
+        for (CacheServerInfo info : information.getCacheServeInfo()) {
```

Comment:
this diff is hard to review due to a lot of unnecessary variable renaming.  I don't feel that renaming `information`  to `mInfo` is such a great improvement to justify all the change&reformatting that was incurred. Might be better just to rename it back.

Reply:
Something we learnt from the book Clean Code that we're trying to implement is more descriptive names. Happy to revert that if it's making it too hard.

Reply:
I'm more inclined to change it to `memberInfo` because it is more descriptive. I understand that naming changes initially are a pain but in the long run the code will benefit greatly from better naming.

Reply:
I'm all for changing to better / more descriptive names.  All I ask is that you consider doing that as a separate PR, or a separate commit within the PR, so that "real" changes can be reviewed easily without having to analyze of ton of diffs to see if the variable name was all that changed or if there was more...